### PR TITLE
Add timeout to github API request

### PR DESCRIPTION
**Pull request by HighCommander4**
This avoids the auto-install process hanging if the github website is unresponsive for some reason.

Fixes the remaining issue in https://github.com/clangd/vscode-clangd/issues/406

### CODE REVIEW

**Review by hokein — APPROVED** [review]
Thanks.

File: src/index.ts [hunk 1]
```diff
@@ -165,7 +165,9 @@ export interface Asset {
 
 // Fetch the metadata for the latest stable clangd release.
 export async function latestRelease(): Promise<Release> {
-  const response = await fetch(githubReleaseURL);
+  const response = await fetch(githubReleaseURL, {
+    timeout: 5000  // milliseconds
```
[the remark this answers is not in the document]
Comment (HighCommander4) [reply src/index.ts]: That's unfortunate, the new API looks quite a bit more verbose from the user's point of view. Anyways, I've updated the patch to use the new API.

**Comment by HighCommander4** [comment]
Returning for review, just to make sure I'm using `AbortController` appropriately. In particular, I'm just letting the `AbortError` propagate instead of catching and re-throwing something else, hope that's ok.

File: src/index.ts [hunk 2]
```diff
@@ -165,12 +165,19 @@ export interface Asset {
 
 // Fetch the metadata for the latest stable clangd release.
 export async function latestRelease(): Promise<Release> {
-  const response = await fetch(githubReleaseURL);
-  if (!response.ok) {
-    console.log(response.url, response.status, response.statusText);
-    throw new Error(`Can't fetch release: ${response.statusText}`);
+  const timeoutController = new AbortController();
```
Comment (hokein) [inline src/index.ts:168]: per the doc, AbortController was added in node v14.17.0 globally.

It requires a >=v14.17.0 node. I think we should be fine -- clangd vscode extension supports vscode v1.65.0 or later, the vscode v1.65 is based on with the electron v13.5.2 which comes with the node v14.17.0, perfect hit!

File: src/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (hokein) [reply src/index.ts]: Agree, it is too verbose (sorry, I didn't meant to ask you to adapt the new API). Anyway the new implementation looks good to me as well.

**Review by hokein — APPROVED** [review]
still looks good.

File: src/index.ts (hunk 2 above)
Comment (HighCommander4) [reply src/index.ts:168]: Thanks for checking! I figured we should avoid introducing a use of a deprecated API if we can avoid it.

**Pull request merged**